wgpu-core/src/command/fill_clear.rs

```diff
@@ -0,0 +1,297 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use std::ops::Range;
+
+#[cfg(feature = "trace")]
+use crate::device::trace::Command as TraceCommand;
+use crate::{
+    command::CommandBuffer,
+    conv,
+    device::all_buffer_stages,
+    hub::{GfxBackend, Global, GlobalIdentityHandlerFactory, Token},
+    id::{BufferId, CommandEncoderId, TextureId},
+    memory_init_tracker::{MemoryInitKind, MemoryInitTrackerAction},
+    resource::{BufferUse, TextureUse},
+    track::TextureSelector,
+};
+
+use hal::command::CommandBuffer as _;
+use thiserror::Error;
+use wgt::{BufferAddress, BufferUsage, Color, ImageSubresourceRange, TextureAspect, TextureUsage};
+
+/// Error encountered while attempting a clear.
+#[derive(Clone, Debug, Error)]
+pub enum FillOrClearError {
```

Comment:
it would be great to come up with a single name for this operation. clear_buffer and clear_image sound good to me. We'd then name this module `clear.rs`, and have the error enum called `ClearError`.

Reply:
agreed! Very messy right now because I sticked so much with the underlying gfx/vulkan api names